File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.java
class Solution {
    public int[] solution(int[] numbers, String direction) {
        int len = numbers.length;
        int[] answer = new int[len];
        if ( direction.equals("right") ) {
            for ( int i = 0; i < len-1; i++ ) {
                answer[i+1] = numbers[i];
            }
            answer[0] = numbers[len-1];
        } else if ( direction.equals("left") ) {
            for ( int i = 1; i < len; i++ ) {
                answer[i-1] = numbers[i];
            }
            answer[len-1] = numbers[0];
        }
        return answer;
    }
}

File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.js
function solution(numbers, direction) {
    var answer = [];
    const len = numbers.length;
    
    const first = numbers[0];
    const last = numbers[len-1];
    
    const middle = numbers.reduce((acc, num, index) => {
        if ( index == 0 || index == len-1 ) return acc;
        
        acc.push(num);
        
        return acc;
    }, []);
    
    if ( direction == "right") {
        answer = [last, first, ...middle];
    } else if ( direction == "left" ) {
        answer = [...middle, last, first];
    }
    return answer;
}